membership-common/src/main/scala/com/gu/okhttp/RequestRunners.scala

```diff
@@ -1,55 +1,58 @@
 package com.gu.okhttp
 
+import com.gu.monitoring.SafeLogging
+import okhttp3.{Call, Callback, OkHttpClient, Request, Response}
+
+import java.io.IOException
 import java.util.concurrent.TimeUnit
-import com.gu.lib.okhttpscala._
-import com.gu.memsub.util.Timing
-import com.gu.monitoring.StatusMetrics
-import okhttp3.{OkHttpClient, Request, Response => OkResponse}
+import scala.concurrent.{ExecutionContext, Future, Promise}
 import scala.concurrent.duration._
-import scala.concurrent.{ExecutionContext, Future}
-import scala.language.higherKinds
-import scalaz.{Kleisli, ReaderT}
 
 /** These are functions from an OkHttpRequest to an M[Response] which are passed into Clients (such as SimpleClient), to determine how they process
   * HTTP requests
   */
 object RequestRunners {
   lazy val client = new OkHttpClient()
-  type FutureHttpClient = Request => Future[OkResponse]
-  type LoggingHttpClient[A[_]] = Request => ReaderT[String, A, OkResponse]
+
+  trait HttpClient[M[_]] {
+    def execute(request: Request): M[Response]
+  }
+
+  class FutureHttpClient(client: OkHttpClient) extends HttpClient[Future] with SafeLogging {
+
+    def execute(request: Request): Future[Response] = {
+      val p = Promise[Response]()
+
+      client
+        .newCall(request)
+        .enqueue(new Callback {
+          override def onFailure(call: Call, e: IOException): Unit = {
+            val sanitizedUrl = s"${request.url().uri().getHost}${request.url().uri().getPath}" // don't log query string
+            logger.warn(s"okhttp request failure: ${request.method()} $sanitizedUrl", e)
+            p.failure(e)
+          }
+
+          override def onResponse(call: Call, response: Response): Unit = {
+            p.success(response)
+          }
+        })
+
+      p.future
+    }
+
+  }
 
   /** Standard no frills run this request and return a response asynchronously A solid choice for the beginner SimpleClient user
-    */
-  def futureRunner(implicit ec: ExecutionContext): Request => Future[OkResponse] =
-    client.execute
+   */
+  def futureRunner(implicit ec: ExecutionContext): FutureHttpClient =
+    new FutureHttpClient(client)
 
   /** Adjusts the standard client used in futureRunner to use a configurable read timeout setting, see:
-    * https://github.com/square/okhttp/wiki/Recipes#timeouts
-    */
-  def configurableFutureRunner(timeout: Duration)(implicit ec: ExecutionContext): Request => Future[OkResponse] = {
+   * https://github.com/square/okhttp/wiki/Recipes#timeouts
+   */
+  def configurableFutureRunner(timeout: Duration)(implicit ec: ExecutionContext): FutureHttpClient = {
     val seconds: Int = timeout.toSeconds.toInt
-    client.newBuilder().readTimeout(seconds, TimeUnit.SECONDS).build().execute
+    new FutureHttpClient(client.newBuilder().readTimeout(seconds, TimeUnit.SECONDS).build())
   }
 
-  /** Logging runner with the default read timeout (10 seconds).
```

Comment:
logging stuff was not used any more, and I didn't want to get my head around ReaderT at this stage